core/domain/collection_domain.py

```diff
@@ -389,6 +453,56 @@ def _convert_collection_contents_v1_dict_to_v2_dict(
         return collection_contents
 
     @classmethod
+    def _convert_collection_contents_v2_dict_to_v3_dict(
+            cls, collection_contents):
+        """Converts from version 2 to 3.
+
+        Adds a skills dict and skill id counter. Gets skills in
+        prerequisite_skills and acquired_skills in nodes, and assigns them
+        integer IDs.
+        """
+        collection_contents['skills'] = {}
+        collection_contents['skills_id_count'] = 0
+
+        def _add_or_get_id(skill):
```

Comment:
Better to separate into two distinct functions.

Also, perhaps simpler to just dump all prereq and acquired skills into a set, then make that into a skill --> id dict and update the lists accordingly? Seems easier to read, since there'd be less need to branch into cases and manually keep track of skill_id_count that way.